lightning/src/ln/channel.rs

```diff
@@ -1946,7 +1946,17 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 
 		let payment_hash = PaymentHash(Sha256::hash(&msg.payment_preimage.0[..]).into_inner());
-		self.mark_outbound_htlc_removed(msg.htlc_id, Some(payment_hash), None).map(|source| source.clone())
+		match self.mark_outbound_htlc_removed(msg.htlc_id, Some(payment_hash), None).map(|source| source.clone()) {
+			Ok(res) => {
+				let mon_info = MonitorUpdateInfo {
+					funding_outpoint: self.funding_txo.unwrap(),
+					latest_monitor_update_id: self.latest_monitor_update_id,
```

Comment:
This doesn't work, you're passing the monitor_update_id of a downstream channel to an upstream one, there is no guarantee they match as downstream/upstream channels are updated asynchronously.

Here downstream/upstream are to be interpreted as the order of HTLC relay, Alice -> Bob -> Caroll. Upstream is Alice-Bob, downstream is Bob-Caroll.

Reply:
Right, I think we might as well just pass a sentinel here - we'd have to store a mapping of closed channels -> last monitor update id in `ChannelManager` otherwise and it doesn't seem worth it - we should just pass `u64::MAX` and note in documentation that "after the channel is closed more updates may occur until all HTLCs have been resolved, possibly using u64::MAX as the update_id"

Reply:
Still, looking more in the logs, I think there is an underlying issue masked by the first monitor id order one.

* in `process_pending_monitor_events`, if we receive a `CommitmentTxBroadcast` for a given channel, we remove it from `channel_state.short_to_id`, thus stopping any off-chain operation on it
* in `claim_funds_from_hop`, using `HTLCPreviousHopData` we fetch the `chan_id` from `channel_state.short_to_id`, if it's missing we return an error

It sound we need to keep a reference to a channel in `channel_state` until all the forward HTLCs corresponding to the incoming HTLCs of this channel have been settled, either onchain or offchain.

Instead of removing channel in `process_pending_monitor_events`, maybe we should mark them irrevocably disabled.